### PR TITLE
Fix adaptive authentication script file execution issue

### PR DESCRIPTION
**Pull request by Shenali-SJ**
### Purpose
This resolves issues encountered when downloads fail during the execution of adaptive.sh and adaptive.bat files.

### Related Issues
https://github.com/wso2/product-is/issues/16158

### CODE REVIEW

File: modules/distribution/src/bin/adaptive.sh [hunk 1]
```diff
@@ -127,7 +127,11 @@ else
   else
     sever_restart_required=true
     echo "Nashorn library not found. Starting to download....."
-    curl https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar -o $CARBON_HOME/repository/components/lib/nashorn-core-$NASHORN_VERSION.jar
+    res_code=$(curl https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar -o $CARBON_HOME/repository/components/lib/nashorn-core-$NASHORN_VERSION.jar)
+    if test "$exit_code" != "0"; then
+      echo "Nashorn library download failed with exit code: $exit_code"
+      exit 1
+    fi
```
Comment (Kanapriya) [inline modules/distribution/src/bin/adaptive.sh:134]: Don't we need to add this changes to the adaptive.bat file?

Comment (Shenali-SJ) [reply modules/distribution/src/bin/adaptive.sh:134]: The related issue was reported for only the shell script. Shall we look into bat files as well?

**Comment by melanisilva** [comment]
Verified the change in Windows OS 
Both scenarios are working as expected, Where the adaptive authentication scripts fail and pass.
success

File: modules/distribution/src/bin/adaptive.bat [hunk 2]
```diff
@@ -155,12 +165,23 @@ if exist "%CARBON_HOME%\repository\components\lib\asm-util-*.jar" (
     call :removeLibrary "ASM Util", "dropins", "%CARBON_HOME%\repository\components\dropins\asm_util_!LOCAL_ASM_VERSION!*.jar" 
     echo Downloading required ASM-Util library : asm-util-%ASM_VERSION%
 	  curl https://repo1.maven.org/maven2/org/ow2/asm/asm-util/%ASM_VERSION%/asm-util-%ASM_VERSION%.jar -o %CARBON_HOME%/repository/components/lib/asm-util-%ASM_VERSION%.jar
+	rem if download failed
```
Comment (ayeshajay) [inline modules/distribution/src/bin/adaptive.bat:168]: Check indentation

**Pull request merged**